File: 10-K-th Smallest Prime Fraction/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> kthSmallestPrimeFraction(vector<int>& arr, int k) {
        vector<pair<vector<int>, double>> fractions;

        for (int i = 0; i < arr.size(); ++i) {
            for (int j = 0; j < arr.size(); ++j) {
                fractions.push_back({{arr[i], arr[j]}, static_cast<double>(arr[i]) / arr[j]});
            }
        }

        sort(fractions.begin(), fractions.end(), [](const auto& a, const auto& b) {
            return a.second < b.second;
        });

        return fractions[k - 1].first;
    }
};

File: 10-K-th Smallest Prime Fraction/solution.py
class Solution:
    def kthSmallestPrimeFraction(self, arr, k):
        fractions = []

        for i in range(len(arr)):
            for j in range(len(arr)):
                fractions.append([[arr[i], arr[j]], arr[i] / arr[j]])

        fractions.sort(key=lambda x: x[1])

        return fractions[k][0]

sol = Solution()
print(sol.kthSmallestPrimeFraction([1, 2, 3, 5], 2))